### PR TITLE
Global Styles: rename `core` origin key to `default` for presets

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
Renames core origin key to default.

I verified global styles continue to work as before.

### CODE REVIEW

File: packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js [hunk 1]
```diff
@@ -215,7 +215,7 @@ const PanelColorGradientSettingsMultipleSelect = ( props ) => {
 
 	const customGradients = useSetting( 'color.gradients.user' );
 	const themeGradients = useSetting( 'color.gradients.theme' );
-	const defaultGradients = useSetting( 'color.gradients.core' );
+	const defaultGradients = useSetting( 'color.gradients.default' );
```
Comment (youknowriad) [inline packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js:218]: I guess this one might be a breaking change? Should we provide an alias or maybe it's fine for this one?

Comment (jorgefilipecosta) [reply packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js:218]: I think it is fine origins were never exposed in the core.

**Review by youknowriad — COMMENTED** [review]
I think I still see a couple places to change:

 - packages/edit-site/src/components/global-styles/use-global-styles-output.js
 - phpunit/class-wp-theme-json-test.php

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by geriux** [comment]
Hey there 👋  I was wondering if there's a way for future changes like these to still support a previous naming? The mobile editor uses a global style endpoint where it gets the colors configuration including `gradients` once this change goes live, current mobile users that have a block-based theme activated won't see the gradients until they upgrade to the next version of the app (which it'll be in a few weeks close to a month) because in the code it'll look for the `core` values which are now `default`.

**Pull request merged**